### ISSUE TITLE
[Feature Request] Core component to select Folder

### ISSUE THREAD

**Issue by mheydasch** [plotly/dash-core-components#639]
It would be really useful to get a core component, such as the upload button to just select a folder. No files directly being uploaded, just getting the path to the folder as an output.

I need this for my application because I do have files in sub folders that need to be uploaded.
These are hundreds of files in tens of sub folders. Therefore it is not possible to use the upload button for this case. Such file architecture is very common in the life sciences.

At the moment I need my user to type in the path by hand such as
/User/user1/Desktop/Experiments/Experiment_1/

A core component enabling the selection of a folder therefore would be very useful.

**Comment by FPRM** [comment]
+1 for the feature

**Comment by esmidev** [comment]
+1

**Comment by PleaseStateTheNatureOfYourInquiry** [comment]
I completely agree. I am running into the same problem. I need to locate a file, get the filename (the Upload component can do that) AND the file path!! I cannot believe that this does not exist in Dash. Seems like the number 1 thing you want to be able to handle in any science related application.

**Comment by gioxc88** [comment]
+1

**Comment by 32nguyen** [comment]
+ 1

**Comment by nbx0** [comment]
+1

**Comment by sceeptex** [comment]
+1

**Comment by j-colas** [comment]
+1 !

**Comment by AimoreRRD** [comment]
Any update on that? That seems a very useful tool that I was expecting dash to have. Any way around this for the moment?

**Comment by mheydasch** [comment]
> Any update on that? That seems a very useful tool that I was expecting dash to have. Any way around this for the moment?
I think this thread might have an idea for something like a work around? But much more it reads to me as this is a functionality that is not intended to be there.
https://community.plotly.com/t/get-file-path-from-upload-component/9934/4

**Comment by daneletourneau** [comment]
+1

**Comment by guigaut** [comment]
+1

**Comment by SebasGarcia08** [comment]
+1

**Comment by JingweiToo2** [comment]
+1

**Comment by shanepalmer1** [comment]
+1 !!

**Comment by linokl** [comment]
+1 really need this feature for science applications

**Comment by SimonKnorke** [comment]
+1

**Comment by christian-oreilly** [comment]
+1

**Comment by fgiaret** [comment]
+1 would be great!

**Comment by gioarma** [comment]
+1 I would really love this feature!
